src/top-issues/top-issues.controller.ts

```diff
@@ -0,0 +1,35 @@
+import { BadRequestException, Body, Controller, Delete, Get, HttpCode, Param, ParseIntPipe, Post, Put, UsePipes } from '@nestjs/common';
+import { TopIssuesService } from './top-issues.service';
+import { CreateTopIssueDto, UpdateTopIssueDto } from './schemas/topIssues.schema';
+import { ZodValidationPipe } from 'nestjs-zod';
+
+@Controller('top-issues')
+@UsePipes(ZodValidationPipe)
+export class TopIssuesController {
+  constructor(private readonly topIssuesService: TopIssuesService) {}
+  
+  @Get()
+  listTopIssues() {
+    return this.topIssuesService.list();
+  }
+
+  @Post()
+  async createTopIssue(@Body() body: CreateTopIssueDto) {
+    return await this.topIssuesService.create(body);
+  }
+
+  @Put(':id')
+  async updateTopIssue(
+    @Param('id', ParseIntPipe) id: number,
+    @Body() body: UpdateTopIssueDto
```

Comment:
Oh.. I see what you did here. Yes, RESTful endpoints would have the id on the url params.. Good.

But it's perfectly acceptable to expect the ID to be on the body as well. Especially given that this is a `PUT` endpoint, and not a `PATCH` endpoint.

The RESTful expectation of `PUT` is that the client is saying to the server, "Whatever is at this URL right now, _replace_ it w/ what I'm sending you in the body, so that the next time I call a `GET` to this location, I'll get the idempotent result of exactly what I sent for the `PUT`", which would hopefully include the `id` property as well.